### PR TITLE
Add ref:vatin test V2

### PR DESCRIPTION
**Pull request by fansanelli**
New version with most of comments addressed

### CODE REVIEW

**Comment by frodrigo** [comment]
Note, you can push on your branch to update the pull request.

File: plugins/TagFix_Vatin.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            y += 2 * (int(vatin[i:i + 1]))
+            if int(vatin[i:i + 1]) >= 5:
+                z = z + 1
+        t = (x + y + z) % 10
+        c = int(vatin[10:11])
+        return c == (10 - t) % 10
+
+    def node(self, data, tags):
+        if "ref:vatin" in tags:
+            x = re.split("^([A-Z]{2})(.+)$", tags["ref:vatin"])
```
Comment (frodrigo) [inline plugins/TagFix_Vatin.py]: Precompile the regex in the init. But here you does not need a regex `if tags["ref:vatin"].startswith('IT')`

Comment (frodrigo) [reply plugins/TagFix_Vatin.py]: and `tags["ref:vatin"]|2:]` for the last part.

Comment (fansanelli) [reply plugins/TagFix_Vatin.py]: Well, consider that any country code (IT, DE, FR,...) followed by a code (alphanum) is valid...
I am not a python expert, how should I pre-compile it?

Comment (frodrigo) [reply plugins/TagFix_Vatin.py]: Not sure what you want to achieve here. But avoid using regex if not required.
You can check the first two are country code like https://docs.python.org/fr/2/library/stdtypes.html?highlight=isalpha#str.isalpha

To compile regex https://docs.python.org/2/library/re.html#re.compile

Comment (fansanelli) [reply plugins/TagFix_Vatin.py]: The idea was to check that the pattern <CountryCode><VAT> is respected, but could achieved in a later revision of this file

**Comment by fansanelli** [comment]
I slept on this. Please wait before merging I want to do restore the check inside an else branch

**Comment by fansanelli** [comment]
I'm quite happy with it, even without the regex. Please let me know if we're done

**Comment by frodrigo** [comment]
Merged. Thank you.
I will ping you when the code will be deployed.

**Comment by fansanelli** [comment]
Thank you so much

**Comment by frodrigo** [comment]
Results are here, please confirm it's OK.
http://osmose.openstreetmap.fr/fr/errors/?item=3260&useDevItem=all
This new item is still hidden.

**Comment by frodrigo** [comment]
Humm, can you try to report more details? like bad length, bad checkesum, checksum is missing, country code is missing...

**Comment by fansanelli** [comment]
Hello,
I've checked some values (first ten or so) and I can confirm all of them should be fixed...
Anyway my code has 2 sub errors... 0 is "invalid format" and 1 "bad Italian partita IVA"

**Comment by frodrigo** [comment]
The display can be improved by using multiple class (self.errors[99999]) and/or a subtitle description (text): https://github.com/osm-fr/osmose-backend/blob/master/plugins/Source.py#L45

**Comment by fansanelli** [comment]
Do you think I can add the texts above or should I create more subclasses?

**Comment by frodrigo** [comment]
You can do both (class, no subclasses)

**Comment by frodrigo** [comment]
@jocelynj please can you enable item 3260?

**Comment by frodrigo** [comment]
Item publicly available now
http://osmose.openstreetmap.fr/en/map/#item=3260&zoom=6&lat=44.72&lon=11.96&level=1%2C2%2C3&tags=&fixable=

**Comment by fansanelli** [comment]
Thanks for the help!

**Pull request merged**